content/blog/2019-01-31-rdhs.md

```diff
@@ -0,0 +1,787 @@
+variables within `travis` itself, which could then be used to create a valid
+set of credentials. 
+
+![](/img/blog-images/2019-01-31-rdhs/travis_env_vars.png)
+
+This worked, however, it meant that I would have to write a lot of the `rdhs`
+functionality to use environment variables that were the user's email and password, 
+which felt wrong and quite clunky. All I wanted was to pass to Travis was a valid
+set of login credentials that would then be used within the tests, much in the same
+way that a user would. To do this I had to learn a bit more about what the `.travis.yml`
+document could actually be used for, because to begin with I had only been using it
+to specify the software language.
+
+Again, Rich pointed me to using `sodium` to create an encrypted version of a valid
+login credentials:
+
+
+```r
+# read in a key from a local file
+key <- sodium::hash(charToRaw(readLines("scripts/key.txt")))
+
+# create a tat with all the necessary login credentials
+zip("rdhs.json.tar",files=c("rdhs.json", "tests/testthat/rdhs.json"))
+
+# read this tar in as binary data 
+dat <- readBin("rdhs.json.tar",raw(),file.size("rdhs.json.tar"))
+
+# encrypt the data using sodium and our key before saving it
+enc <- sodium::data_encrypt(msg = dat,key = key)
+saveRDS(enc,"rdhs.json.tar.enc")
+```
+
+This could then be included in the GitHub repository, and I could then
+set up the key as a Travis environment variable to decrypt it. This decryption 
+step could then be written within my `.travis.yml` file, and would mean that all
+my tests had access to my login credentials in a secure way. 
+
+### Acknowledgements and Final Thoughts
+
+Firstly, I want to thank [Anna Krystalli](https://github.com/annakrystalli) for handling
+the review, and for being incredibly patient throughout, especially at the end as we
+were fixing the last authentication bug. Also many thanks to [Lucy McGowan](https://github.com/LucyMcGowan)
+and [Duncan Gillespie](https://github.com/dosgillespie) for taking the time to 
+review the package and for their input, which led to lots of improvements (and also
+linking the `add_line` function from `httr` was seriously helpful, and I've used 
+that function in lots of other my other work now). I also wanted to more broadly thank 
+the review process as a whole. Having the option to discuss the package and needed 
+solutions with the reviewers within a GitHub issues system is fantastic. It made the process 
+personal and was substantially improved over review processes I have had at academic journals.
+Lastly, another big thank you [Jeff Eaton](https://github.com/jeffeaton) and 
+[Rich FitzJohn](https://github.com/richfitz), and also to the infectious
+disease epidemiology department at Imperial for providing a lot of really helpful 
+guinea pig testing of the numerous iterations of `rdhs`.
+
+---
+
+### Options to Contribute
+
+There are a few things that would be great to add in the future.
```

Comment:
For visual variety, you might put these in a numbered list rather than saying "first, second...", but no obligation to do this